pre-docs/inventory_version2.md

```diff
@@ -180,6 +180,77 @@ groups:
             hostname: facts.networking.interfaces.en0.ipaddress
 ```
 
+#### Task
+
+The Task plugin will run a task on `localhost` to lookup configuration
+information or target lists for inventory..
+
+For both use cases the plugin accepts two keys:
+* `task`: The task to run.
+* `parameters`: The parameters to pass to the task.
+
+example:
+
+```
+---
+version: 2
+targets:
+  - _plugin: task
+    task: my_json_file::targets
+    parameters:
+      # These parameters are specific to the task
+      file: /etc/targets/data.json
+      environment: production
+      app: my_app
+config:
+  ssh:
+    password:
+      _plugin: task
+      task: my_db::secret_lookup
+      parameters:
+        # These parameters are task specific
+        key: ssh_password
+```
+
+##### Inventory Config Tasks
+
+To lookup configuration information for the inventory return a `config` key in
+the task containing the data that should be using in place of the plugin entry.
+The value of config can be any form of data that is appropriate for the
+specific location in config. where it is used.
+
```

Comment:
For example, this task reads in a JSON file, gets a secret key, and returns <output>.